service/importer/definition.go

```diff
@@ -1,108 +1,111 @@
 package importer
 
+// ConfigurationDependencyKey is the reserved key of the service's configuration in the dependencies array.
+const ConfigurationDependencyKey = "service"
+
 // ServiceDefinition represents MESG services configurations.
 type ServiceDefinition struct {
 	// Name is the service name.
-	Name string `yaml:"name" json:"name,omitempty"`
+	Name string `yaml:"name" json:"name,omitempty" validate:"required,printascii,min=1"`
 
 	// Sid is the service id. It must be unique.
-	Sid string `yaml:"sid" json:"sid,omitempty"`
+	Sid string `yaml:"sid" json:"sid,omitempty" validate:"printascii,max=39"`
 
 	// Description is service description.
-	Description string `yaml:"description" json:"description,omitempty"`
+	Description string `yaml:"description" json:"description,omitempty" validate:"printascii"`
 
 	// Tasks are the list of tasks that service can execute.
-	Tasks map[string]*Task `yaml:"tasks" json:"tasks,omitempty"`
+	Tasks map[string]*Task `yaml:"tasks" json:"tasks,omitempty" validate:"dive,keys,printascii,endkeys,required"`
 
 	// Events are the list of events that service can emit.
-	Events map[string]*Event `yaml:"events" json:"events,omitempty"`
+	Events map[string]*Event `yaml:"events" json:"events,omitempty" validate:"dive,keys,printascii,endkeys,required"`
 
 	// Dependencies are the Docker containers that service can depend on.
-	Dependencies map[string]*Dependency `yaml:"dependencies" json:"dependencies,omitempty"`
+	Dependencies map[string]*Dependency `yaml:"dependencies" json:"dependencies,omitempty" validate:"dive,keys,printascii,ne=service,endkeys,required"`
 
 	// Configuration is the Docker container that service runs inside.
 	Configuration *Dependency `yaml:"configuration" json:"configuration,omitempty"`
 
 	// Repository holds the service's repository url if it's living on
 	// a Git host.
-	Repository string `yaml:"repository" json:"repository,omitempty"`
+	Repository string `yaml:"repository" json:"repository,omitempty" validate:"omitempty,uri"`
 }
 
 // Event describes a service task.
 type Event struct {
 	// Name is the name of event.
-	Name string `yaml:"name" json:"name,omitempty"`
+	Name string `yaml:"name" json:"name,omitempty" validate:"printascii"`
 
 	// Description is the description of event.
-	Description string `yaml:"description" json:"description,omitempty"`
+	Description string `yaml:"description" json:"description,omitempty" validate:"printascii"`
 
 	// Data holds the input parameters of event.
-	Data map[string]*Parameter `yaml:"data" json:"data,omitempty"`
+	Data map[string]*Parameter `yaml:"data" json:"data,omitempty" validate:"required,dive,keys,printascii,endkeys,required"`
 }
 
 // Dependency represents a Docker container and it holds instructions about
 // how it should run.
 type Dependency struct {
 	// Image is the Docker image.
-	Image string `yaml:"image" json:"image,omitempty"`
+	Image string `yaml:"image" json:"image,omitempty" validate:"printascii"`
```

Comment:
```suggestion
	Image string `yaml:"image" json:"image,omitempty" validate:"printascii,min=1"`
```

Reply:
Hum, so it makes things bit complicated. In that case we should validate only dependencies if image is not empty and here required must be dropped